### PR TITLE
DOC: improved the scatter method

### PR DESCRIPTION
**Pull request by DaniGate**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [x] PR title is "DOC: update the <your-function-or-method> docstring"
- [x] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [x] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [x] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```


################################################################################
################## Docstring (pandas.DataFrame.plot.scatter)  ##################
################################################################################

A scatter plot with point size *s* and color *c*.

The coordinates of each point *x,y* are defined by two dataframe
columns and filled circles are used to represent each point.

Parameters
----------
x : column name or column position
    Horizontal and vertical coordinates of each point.
y : column name or column position
    Vertical coordinates of each point.
s : scalar or array_like, optional
    Size of each point.
c : label, column name or column position, optional
    Color of each point.
kwds : optional
    Keyword arguments to pass on to :py:meth:`pandas.DataFrame.plot`.

Returns
-------
axes : matplotlib.AxesSubplot or np.array of them

See Also
--------
matplotlib.pyplot.scatter : scatter plot using multiple input data
    formats.

Examples
--------

.. plot::
    :context: close-figs

    >>> from sklearn.datasets import load_iris
    >>> iris = load_iris()
    >>> df = pd.DataFrame(iris.data[:,:2],
    ...                   columns=iris.feature_names[:2])
    >>> df['species'] = load_iris().target
    >>> f = df.plot.scatter(x='sepal length (cm)',
    ...                     y='sepal width (cm)',
    ...                     c='species',
    ...                     colormap='viridis')

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.DataFrame.plot.scatter" correct. :)

```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

### CODE REVIEW

File: pandas/plotting/_core.py [hunk 1]
```diff
@@ -2852,22 +2852,48 @@ def pie(self, y=None, **kwds):
 
     def scatter(self, x, y, s=None, c=None, **kwds):
         """
-        Scatter plot
+        A scatter plot with point size *s* and color *c*.
```
Comment (TomAugspurger) [inline pandas/plotting/_core.py]: How about the first line is "A scatter plot of two columns in the DataFrame."

File: pandas/plotting/_core.py [hunk 2]
```diff
@@ -2852,22 +2852,48 @@ def pie(self, y=None, **kwds):
...
+
+        The coordinates of each point *x,y* are defined by two dataframe
```
Comment (TomAugspurger) [inline pandas/plotting/_core.py]: Parameter names should be in singel backticks.

File: pandas/plotting/_core.py [hunk 3]
```diff
@@ -2852,22 +2852,48 @@ def pie(self, y=None, **kwds):
...
+        columns and filled circles are used to represent each point.
 
         Parameters
         ----------
-        x, y : label or position, optional
-            Coordinates for each point.
+        x : column name or column position
```
Comment (TomAugspurger) [inline pandas/plotting/_core.py]: I think `label or position` is OK. You can remove optional, as both `x` and `y` are required.

File: pandas/plotting/_core.py [hunk 4]
```diff
@@ -2852,22 +2852,48 @@ def pie(self, y=None, **kwds):
...
+        matplotlib.pyplot.scatter : scatter plot using multiple input data
+            formats.
+
+        Examples
+        --------
+
+        .. plot::
+            :context: close-figs
+
+            >>> from sklearn.datasets import load_iris
```
Comment (TomAugspurger) [inline pandas/plotting/_core.py]: We can't import sklearn here, since it isn't installed.

Comment (TomAugspurger) [reply pandas/plotting/_core.py]: Maybe just make a small sample dataset instead.

Comment (DaniGate) [reply pandas/plotting/_core.py]: It passed all checks and test and the plot was shown in the docs. But ok, I will write down the dataframe explicitely.

File: pandas/plotting/_core.py (hunk 3 above)
Comment (DaniGate) [reply pandas/plotting/_core.py]: But the description of the argument `c` specifies column name or column position. My intention was to make it consistent

File: pandas/plotting/_core.py [hunk 5]
```diff
@@ -2852,22 +2852,46 @@ def pie(self, y=None, **kwds):
 
     def scatter(self, x, y, s=None, c=None, **kwds):
         """
-        Scatter plot
+        A scatter plot with point size `s` and color `c`.
```
Comment (dukebody) [inline pandas/plotting/_core.py]: To be consistent with the rest of the docs from the chapter I'd start this with an infinitive verb like "Create a scatter plot", "Generate a scatter plot", "Make a scatter plot"...

**Comment by dukebody** [comment]
FTR this is pandas.DataFrame.plot.scatter.

File: pandas/plotting/_core.py [hunk 6]
```diff
@@ -2852,22 +2852,46 @@ def pie(self, y=None, **kwds):
...
+
+        See Also
+        --------
+        matplotlib.pyplot.scatter : scatter plot using multiple input data
+            formats.
+
+        Examples
+        --------
+
+        .. plot::
```
Comment (dukebody) [inline pandas/plotting/_core.py]: Can you add some text above the code explaining what are you doing here?

File: pandas/plotting/_core.py [hunk 7]
```diff
@@ -2852,22 +2852,46 @@ def pie(self, y=None, **kwds):
...
+            :context: close-figs
+
+            >>> df = pd.DataFrame([[5.1, 3.5, 0], [4.9, 3.0, 0], [7.0, 3.2, 1],
+            ...                    [6.4, 3.2, 1], [5.9, 3.0, 2]],
+            ...                   columns = ['length', 'width', 'species'])
+            >>> f = df.plot.scatter(x='length',
```
Comment (dukebody) [inline pandas/plotting/_core.py]: If what is returned is an axis, better to use `ax = ...` here.

File: pandas/plotting/_core.py [hunk 8]
```diff
@@ -2852,22 +2852,46 @@ def pie(self, y=None, **kwds):
...
+            ...                     y='width',
+            ...                     c='species',
+            ...                     colormap='viridis')
```
Comment (dukebody) [inline pandas/plotting/_core.py]: Can you add at least one more example with different parameters to see the differences?

**Review by dukebody — CHANGES_REQUESTED** [review]
Please add the type of the parameters and don't mention them in the description and extended description.

File: pandas/plotting/_core.py (hunk 3 above)
Comment (dukebody) [reply pandas/plotting/_core.py]: There is some confusion about using "label or position" as type. In other PR @datapythonista suggests using real types like "int or str" as the argument type, and mention it's the column name or position in the description of the parameter on next line.

Comment (datapythonista) [reply pandas/plotting/_core.py]: yes, my mistake, I realized later that `label` could be any type, I guess that's why `label or position` is being used.

File: pandas/plotting/_core.py [hunk 9]
```diff
@@ -2852,22 +2852,46 @@ def pie(self, y=None, **kwds):
...
-        x, y : label or position, optional
-            Coordinates for each point.
+        x : column name or column position
+            Horizontal coordinates of each point.
+        y : column name or column position
+            Vertical coordinates of each point.
         s : scalar or array_like, optional
             Size of each point.
-        c : label or position, optional
+        c : label, column name or column position, optional
```
[the remark this answers is not in the document]
Comment (TomAugspurger) [reply pandas/plotting/_core.py]: I think valid options are matplotlib's + a column label, which extracts the array and passes it to matplotlib.

from plt.scatter:

```
c : color, sequence, or sequence of color, optional, default: 'b'
    `c` can be a single color format string, or a sequence of color
    specifications of length `N`, or a sequence of `N` numbers to be
    mapped to colors using the `cmap` and `norm` specified via kwargs
    (see below). Note that `c` should not be a single numeric RGB or
    RGBA sequence because that is indistinguishable from an array of
    values to be colormapped.  `c` can be a 2-D array in which the
    rows are RGB or RGBA, however, including the case of a single
    row to specify the same color for all points.
```

You'll to verify that. Check if / when a colorbar is added.

File: pandas/plotting/_core.py [hunk 10]
```diff
@@ -2852,22 +2852,46 @@ def pie(self, y=None, **kwds):
...
             Color of each point.
-        `**kwds` : optional
+        kwds : optional
```
[the remark this answers is not in the document]
Comment (TomAugspurger) [reply pandas/plotting/_core.py]: And remove the `: optional`. Just `**kwds`.

File: pandas/plotting/_core.py (hunk 8 above)
Comment (TomAugspurger) [reply pandas/plotting/_core.py]: Can you check if we recommend `cmap` or `colormap`?

Comment (DaniGate) [reply pandas/plotting/_core.py]: @TomAugspurger They are both accepted but the one described in pandas.DataFrame.plot is colormap.

File: pandas/plotting/_core.py [hunk 11]
```diff
@@ -2852,22 +2852,47 @@ def pie(self, y=None, **kwds):
 
     def scatter(self, x, y, s=None, c=None, **kwds):
         """
-        A scatter plot with point size `s` and color `c`.
+        Create a scatter plot with varying marker point size and color.
 
-        The coordinates of each point `x,y` are defined by two dataframe
-        columns and filled circles are used to represent each point.
+        The coordinates of each point are defined by two dataframe columns and
```
Comment (dukebody) [inline pandas/plotting/_core.py]: Muuuuch better!!!!!!

File: pandas/plotting/_core.py [hunk 12]
```diff
@@ -2852,22 +2852,47 @@ def pie(self, y=None, **kwds):
...
-        x : column name or column position
-            Horizontal coordinates of each point.
-        y : column name or column position
-            Vertical coordinates of each point.
-        s : scalar or array_like, optional
-            Size of each point.
-        c : label, column name or column position, optional
-            Color of each point.
-        kwds : optional
+        x : int, str
```
Comment (dukebody) [inline pandas/plotting/_core.py]: I think this should be `int or str` instead.

File: pandas/plotting/_core.py [hunk 13]
```diff
@@ -2852,22 +2852,47 @@ def pie(self, y=None, **kwds):
...
+        y : int, str
+            The column name or column position to be used as vertical
+            coordinates for each point.
+        s : scalar, array_like, optional
+            The size of each point. Possible values are:
+
+            - A single scalar so all points have the same size.
+
+            - A sequence of scalars, which will be used for each point's size
+            recursively. For intance [2,14] all points will be size 2 or 14,
```
Comment (dukebody) [inline pandas/plotting/_core.py]: "For instance, using ``[2, 14]`` all points will be of size ...". Typo in "instance", the rest is a proposal to make it more prose-English.

File: pandas/plotting/_core.py [hunk 14]
```diff
@@ -2852,22 +2852,47 @@ def pie(self, y=None, **kwds):
...
+            alternatively.
+
+        c : str, int, array_like, optional
+            The color of each point. Possible values are:
+
+            - A single color string referred to by name, RGB or RGBA code,
```
Comment (dukebody) [inline pandas/plotting/_core.py]: Have you verified if these bullet points render nicely in the final HTML? I'm not good at restructured text so I tend to be wary about these things. :)

File: pandas/plotting/_core.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (dukebody) [reply pandas/plotting/_core.py]: @TomAugspurger I believe we have used `**kwds : optionals` in other similar docstrings. Probably we should come up with an agreement on how to document the `**kwds` parameter in these cases.

**Comment by dukebody** [comment]
This is almost done I believe, just a couple of style fixes and IMO it will be good to merge. :)

File: pandas/plotting/_core.py [hunk 15]
```diff
@@ -2881,17 +2906,22 @@ def scatter(self, x, y, s=None, c=None, **kwds):
...
         --------
+        Let's see how to draw a scatter plot using coordinates and color from
+        the values in three DataFrame columns.
 
         .. plot::
             :context: close-figs
 
             >>> df = pd.DataFrame([[5.1, 3.5, 0], [4.9, 3.0, 0], [7.0, 3.2, 1],
             ...                    [6.4, 3.2, 1], [5.9, 3.0, 2]],
             ...                   columns = ['length', 'width', 'species'])
```
Comment (dukebody) [inline pandas/plotting/_core.py]: Can you remove spaces between the equal sign? `columns=['length...`

File: pandas/plotting/_core.py (hunk 14 above)
Comment (DaniGate) [reply pandas/plotting/_core.py]: You need the blank lines to properly render the bullet points ;)
screen shot 2018-03-14 at 09 59 01

**Comment by TomAugspurger** [comment]
Thanks @DaniGate and @dukebody .

Fixed the line length issue.

**Comment by TomAugspurger** [comment]
OK, also split the examples in two to be a little clearer. Merging, thanks1
